tests/spread/integration/git/task.yaml

```diff
@@ -0,0 +1,157 @@
+summary: Integration tests for git
+
+environment:
+  EXTRA_SLICES: "" # No extra slices by default.
+  # test git_bins
+  RUN/bins: test_bins
+  SLICE/bins: git_bins
+  # test git_http-support
+  RUN/http_support: test_http_support
+  SLICE/http_support: git_http-support
+  EXTRA_SLICES/http_support: ca-certificates_data
+  # test git_ftp-support
+  RUN/ftp_support: test_ftp_support
+  SLICE/ftp_support: git_ftp-support
+  # test git_daemon
+  RUN/daemon_support: test_daemon_support
+  SLICE/daemon_support: git_daemon
+
+execute: |
+
+  git_rootfs="/"
+  function chroot_git(){
+    # execute git within chroot. We use -C to cd since
+    # we have no $SHELL to navigate with. 
+    repo="$(realpath -e --relative-to="$git_rootfs" "$PWD")"
+    chroot "$git_rootfs" /usr/bin/git -C "$repo" "$@"
+  }
+
+  function setup(){
+    mkdir -p "$1/dev/"
+    mkdir -p "$1/etc/"
+
+    # fake essentials
+    touch "$1/dev/null"
+    head -c 500 /dev/urandom > "$1/dev/urandom"
+    cp /etc/resolv.conf $1/etc/
+    mkdir "$1/root"
+
+    # configure
+    git_rootfs="$1"
+    pushd "$1"
+
+  }
+
+  function cleanup(){
+    popd
+  }
+
+  function test_bins(){
+    chroot_git init
+    chroot_git config user.email "root@localhost"
+    chroot_git config user.name "Test Runner"
+
+    echo "Test" > "$1/test.txt"
+    chroot_git add test.txt
+    chroot_git commit -m 'test commit'
+
+    chroot_git --no-pager status
+    chroot_git --no-pager log
+
+    chroot_git blame -L 1,1 test.txt
+    [[ $(chroot_git ls-files) == "test.txt" ]]
+
+    chroot_git checkout -b test
+
+    chroot_git --no-pager diff master
+    chroot_git branch
```

Comment:
if you move this after the `checkout` in the next line, you can grep the output of `chroot_git branch` to make sure it is equal to "master".

Reply:
These are nice changes, but can we do this in another PR? It might be out of scope as git was previously merged into Jammy.